module5/Publisher.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+from typing import List
+
+from module5.ArabicText import ArabicText
+from module5.InputException import InputException
+from module5.News import News
+from module5.PrivateAd import PrivateAd
+from module5.Publication import Publication
+
+
+class Publisher:
+
+    @staticmethod
+    def get_publication_type() -> str:
+        pub_type = input("Welcome to Publisher!\n"
+                         "Please enter publication one of the follow type: News/Private Ad/Arabic Text: ")
+        if pub_type not in Publication.PUBLICATION_TYPES:
+            raise InputException("Publication type is not supported.")
+        return pub_type
+
+    @staticmethod
+    def get_publication_text() -> List[str]:
+        row_count = int(input("Count of text row: "))
+        if row_count <= 0:
+            raise InputException("Row count should be more than 0.")
```

Comment:
Maybe it is worth to extend number of exceptions but give them more appropriate names specifically for particular issues

Reply:
What the reason of many Expansion classes with different name if all of them related to one functionality and haven't different logic within? Only drop user input part with related message.